OVERVIEW.md

```diff
@@ -0,0 +1,200 @@
+# Introduction and Overview
+
+## High-Level: What does it do and why do I need it?
+
+> TODO
+
+## Concepts
+
+* Andrew: Reuse and update "Introductions" section from [top level README](readme.md)
+
+
+### Built-in Rule Types
+
+The CloudDQ framework defines 5 basic types of rules:
+
+* `NOT_NULL`
+* `NOT_BLANK`
+* `REGEX`
+* `CUSTOM_SQL_EXPRESSION`
+* `CUSTOM_SQL_STATEMENT`
+
+Of these rules, the first four support row-level validation and the last one supports set-level validation. We will begin by explaining this difference before proceeding to explain each rule type in detail.
+
+#### Row-level Versus Set-level Validation
+
+Row-level validation checks attributes of individual rows or elements of the data: For example whether the value is in a given range, or whether all fields are internally consistent. There are also validation checks that cannot be performed by just inspecting one row or element. We call this "set-level validation". Possibly the simplest example of set-level validation is a check on the total row count.
+
+An important difference between row-level and set-level validation is that in the case of set-level validation, in general, one cannot say which rows cause a check to fail. Also, in some cases it's not possible to say how many rows fail the check. With row-level validation on the other hand, we can connect each validation error to a specific row. This difference in reflected in the `CloudDQ` output. For row-level validation, `CloudDQ` reports the number of rows validated, how many were successful, how many produced errors and how many were null. For set-level checks, `CloudDQ` only provides how many rows the check was run on, how many errors were generated, plus an overall success or failure flag.
+
+The number of errors from set-level validation may or may not have a relation to a number of rows, depending on how the check is coded. For example, for a check on uniqueness, the number of errors may equal the number of duplicated rows, the number of elements that are duplicated, or something else entirely.
+
+As mentioned, of the built-in rule types, only `CUSTOM_SQL_STATEMENT` is intended for set-level validation. That means that any check implemented using this rule type will produce output that can be expected from set-level validation. This is independent of the nature of the check -- it's certainly feasible to implement row-level checks using rules of the `CUSTOM_SQL_STATEMENT` type.
+
+
+#### Rule Type `NOT_NULL`
+
+Violated if any row in the given column contains null.
+
+The rule is applied to a column in a table (an “entity”)  through a rule binding.
```

Comment:
OK. Dropping the sentence (here and below) because what I wanted to say is we link the rule to the data it's applied to through a rule binding, and just saying "the rule is applied to a column in table" doesn't add anything.